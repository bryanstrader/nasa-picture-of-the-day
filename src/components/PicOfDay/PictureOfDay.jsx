import React from 'react';
import {
    Segment,
    Image,
    Message,
    Dimmer,
    Loader,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';

class PictureOfDay extends React.Component {
    componentDidMount() {
        const { onGetPictureOfDay } = this.props;
        onGetPictureOfDay();
    }

    render() {
        const {
            podDetails,
            error,
            isLoading,
            showTitle,
        } = this.props;
        return (
            <Segment compact className="pod">
                {
                    error
                    && (
                        <Message>
                            NASA Pic of the day unavailable :(
                        </Message>
                    )
                }
                <Image
                    className="podImage"
                    src={podDetails.url}
                    alt={podDetails.title}
                />
                {showTitle && <i>{podDetails.title}</i>}
                <Dimmer active={isLoading} inverted>
                    <Loader />
                </Dimmer>
            </Segment>
        );
    }
}

PictureOfDay.defaultProps = {
    showTitle: true,
    error: '',
    isLoading: false,
};

PictureOfDay.propTypes = {
    podDetails: PropTypes.instanceOf(Object).isRequired,
    onGetPictureOfDay: PropTypes.func.isRequired,
    isLoading: PropTypes.bool,
    error: PropTypes.string,
    showTitle: PropTypes.bool,
};

export default PictureOfDay;
