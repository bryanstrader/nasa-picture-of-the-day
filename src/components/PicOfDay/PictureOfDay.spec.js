import renderer from 'react-test-renderer';
import PictureOfDay from './PictureOfDay';
import { sampleResponse } from '../../utils/mockData';

describe('PictureOfDay Component', () => {
    test('matches snapshot', () => {
        const tree = renderer.create(<PictureOfDay podDetails={sampleResponse} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
