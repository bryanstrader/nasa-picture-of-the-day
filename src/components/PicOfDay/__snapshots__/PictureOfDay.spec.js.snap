// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PictureOfDay Component matches snapshot 1`] = `
<div
  className="ui segment"
>
  <h2>
    NASA Picture of Day
  </h2>
  <img
    alt="Arches Across an Arctic Sky"
    className="ui image"
    src="https://apod.nasa.gov/apod/image/2101/ArcticSky_Cobianchi_1080.jpg"
  />
  <i>
    Arches Across an Arctic Sky
  </i>
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui loader"
      />
    </div>
  </div>
</div>
`;
